bpf/lib/lb.h

```diff
@@ -913,10 +913,15 @@ static __always_inline int __lb4_rev_nat(struct __ctx_buff *ctx, int l3_off, int
 {
 	__be32 old_sip, new_sip, sum = 0;
 	int ret;
+	void *data, *data_end;
+	struct iphdr *ip4;
 
 	cilium_dbg_lb(ctx, DBG_LB4_REVERSE_NAT, nat->address, nat->port);
 
-	if (nat->port) {
+	if (!revalidate_data(ctx, &data, &data_end, &ip4))
+		return DROP_INVALID;
```

Comment:
I don't think we need `data` or `data_end` here, so if I understand correctly we simply want `ip4` header pointer. Would it be possible to pass it down from the parent function instead, to avoid revalidating here? I don't think revalidation should be necessary.

Reply:
changed to pass a bool to indicate has_l4_header everywhere.